src/main/kotlin/pt/up/fe/ni/website/backend/model/Role.kt

```diff
@@ -26,15 +32,22 @@ class Role(
     @JsonProperty(required = true)
     var isSection: Boolean,
 
-    @JoinColumn
-    @ManyToMany
-    var accounts: List<@Valid Account> = emptyList(),
-
-    @JoinColumn
-    @OneToMany
-    var associatedActivities: List<@Valid PerActivityRole> = emptyList(),
+    @ManyToMany(mappedBy = "roles")
+    @JsonIgnore // TODO: Decide if we want to return accounts (or IDs) by default
+    @OnDelete(action = OnDeleteAction.CASCADE) // Remove relationship, since this is the non-owner side
```

Comment:
Using delete cascade, in this case, doesn't mean that deleting a Role will delete the accounts associated with it?

Reply:
This confused me at first too and I actually had that behaviour incorrectly when I first implemented this. That happened because I was using `CascadeType.REMOVE` which is different from `OnDeleteAction.CASCADE`. The `@OnDelete` will delete the table associated with the relationship instead of the account. I believe I wrote a test for that scenario.

This was an article that helped me understand this: https://rogerkeays.com/jpa-cascadetype-remove-vs-hibernate-ondelete

Reply:
Ok, that makes sense, since this is a many to many relationship, i didn't notice that at first ahah.